src/components/TrackNewExercise.tsx

```diff
@@ -33,6 +48,15 @@ export function TrackNewExercise(): JSX.Element {
   const [posting, setPosting] = useState(false)
   const [pullRequestUrl, setPullRequestUrl] = useState('')
 
+  const prepopulate = props.location.state
+
+  if (prepopulate) {
+    setConcepts(prepopulate.concepts || '')
```

Comment:
This will happen _each render_ of `TrackNewExercise`, if `prepopulate` is true. You probably wanted:

```typescript
useEffect(() => {
  if (!prepopulate) {
    return
  }
  
  setPrerequisites((prev) => prev || prepopulate.prerequisites || '')
  ...

}, [prepopulate, setConcepts, setPrerequisites, setOutOfScope, setLearningObjectives])
```

This also ensures you don't override the already set values. Dunno if that matters.